app/src/main/java/fr/free/nrw/commons/category/CategoryEditHelper.java

```diff
@@ -62,16 +62,23 @@ private Observable<Boolean> addCategory(Media media, List<String> categories,
         final String wikiText) {
         Timber.d("thread is category adding %s", Thread.currentThread().getName());
         String summary = "Adding categories";
-
         final StringBuilder buffer = new StringBuilder();
-
-        final String wikiTextWithoutCategory
-            = wikiText.substring(0, wikiText.indexOf("[[Category"));
-
-        if (categories != null && categories.size() != 0) {
+        final String wikiTextWithoutCategory;
+        if (wikiText.contains("Uncategorized")) {
```

Comment:
As the algorithm is becoming a bit complex, would you mind adding a `//` comment explaining it, so that readers get a quick understanding? Thanks a lot!

Reply:
Thanks! Would you mind rewording to `If the picture was uploaded without a category, the wikitext will contain "Uncategorized" instead of "[[Category"`